### ISSUE TITLE
Private API

### ISSUE THREAD

**Issue by cymp** [lvcn1/ws-client-examples#1]
Hello,

Do you plan to deilver private API endpoints (place/cancel orders, balance, ...)

**Comment by dkpostwww** [comment]
Hello.
Private API is our next milestone.
If you will write what you Really need, it is most likely to be done first.

**Comment by cymp** [comment]
Hello,

In the meantime, your support informed me that they accepted me as a member of a pilot testing group of API websockets Alpha-version. 

Most important features for the private API would be:
- place order
- cancel order
- user balances
- get all active orders
- move order (would allow to cancel and replace the order atomically, thus sparing one API call)

**Comment by dkpostwww** [comment]
Hello,
I am not sure about "move order". Making it "really atomic" is a too big deal, but the "cancel, if cancel is not failed then try to place new" can be done.

**Comment by cymp** [comment]
This would be enough.

**Comment by jsveto** [comment]
@cymp it is under beta-v2 branch.

**Comment by jsveto** [comment]
@dkpostwww There is API key/secret commited under python examples in beta-v2 branch. If it belongs to real account with funds, you should disable it ASAP. i would have sent you a private msg, but unfortunately, github disabled that option.

Regards,
jsveto

**Comment by dkpostwww** [comment]
2cymp: switch to beta-v2 branch and you will see
2jsveto: I hope we will release first version today. Just with "put"/"cancel", but "replace" and other api's will be soon.

2jsveto: Thank you for warning, this key is not even from test sandbox, it is from dev sandbox ;)

**Comment by dkpostwww** [comment]
@cymp , @jsveto : first version is released, you can use it ;). It is now in master branch.

**Comment by cymp** [comment]
Thanks, I will work on it and give you feedback tomorrow.

**Comment by jsveto** [comment]
great. ill try to implement it today. On HTTP request, there is rate limit of 1req/2s. Does this limit apply on websocket requests as well?

**Comment by cymp** [comment]
REST limits and WS limits are the same yes.

**Comment by jsveto** [comment]
@cymp When can we expect java examples?

**Comment by cymp** [comment]
As written on the main page: 

> You can find example clients for Python and JavaScript in this repo (**simple clients for Java and C# coming soon**).

**Comment by jsveto** [comment]
@dkpostwww Finally got it working in java. I have one more question regarding Login ttl.

Does Login expire after ttl time?

My example:
Login with ttl 10s
put_trade request (OK)
sleep 20s
put_trade request (NOK - No response from server)

If i increase ttl time to 20s+ second requests succeeds as well. How can i set Login ttl to last throughout whole websocket session?

**Comment by dkpostwww** [comment]
@jsveto sorry, it is a bug with login. On next update (next week, sorry) we will fix it. For now - set ttl to MAX_INT, it will be enough until next week ;)

**Comment by dkpostwww** [comment]
@jsveto @cymp **For now** limits for REST and WS are the same, but we will encrease WS limits soon, for sure.

**Comment by michaeldim** [comment]
A cancel all orders request would be helpful.  It could be a separate request or orderId could be made optional on CancelLimitOrderRequest.

**Comment by cymp** [comment]
@dkpostwww do you still plan to implement the "cancel, if cancel is not failed then try to place new" ?

**Comment by dkpostwww** [comment]
@cymp yes, it is in our plans. This issue will be closed only when it will be done.
